From 12f9630486d50de9793244b14d9cacbab4c023fd Mon Sep 17 00:00:00 2001
From: Paul Thomas <pthomas8589@gmail.com>
Date: Tue, 23 Jun 2020 14:49:39 -0400
Subject: [PATCH 1421/1852] clk: clock-wizard: Add ascii art block diagram of
 MMCM

commit bf721ce259454fd710d31414a0aa41be5dc18620 from
https://github.com/Xilinx/linux-xlnx.git

This block diagram relates the members of 'struct clk_wzrd'
with the actual hardware chain.

Signed-off-by: Paul Thomas <pthomas8589@gmail.com>
Reviewed-by: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/clk/clk-xlnx-clock-wizard.c | 28 +++++++++++++++++++++++++++-
 1 file changed, 27 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/clk-xlnx-clock-wizard.c b/drivers/clk/clk-xlnx-clock-wizard.c
index 99c2a95ff498..806e1178e4f1 100644
--- a/drivers/clk/clk-xlnx-clock-wizard.c
+++ b/drivers/clk/clk-xlnx-clock-wizard.c
@@ -82,7 +82,33 @@ struct clk_wzrd_vco_hints {
 	bool valid_rate;		/* indicates the hints calc is done */
 };
 
-/**
+/*
+ *         MMCM Block Diagram
+ *
+ *         +----------------+  +-----------------+
+ * input ->| vco_clk_div_hw |->| vco_clk_mul_hw  |--+
+ * rate    | (int divide)   |  | (frac multiply) |  |
+ *         +----------------+  +-----------------+  |
+ *                                                  |
+ *      +--------------------------------VCO-rate---+
+ *      |
+ *      |  +----------------+
+ *      +->| clkout[0]      |-> output0 rate
+ *      |  | (frac divide)  |
+ *      |  +----------------+
+ *      |
+ *      |  +----------------+
+ *      +->| clkout[1]      |-> output1 rate
+ *      |  | (int divide)   |
+ *      |  +----------------+
+ *      |
+ *     ...
+ *      |
+ *      |  +----------------+
+ *      +->| clkout[1]      |-> output6 rate
+ *         | (int divide)   |
+ *         +----------------+
+ *
  * struct clk_wzrd - Clock wizard private data structure
  *
  * @clk_data:		Clock data
-- 
2.31.1

